wandb/sdk/launch/agent/agent.py

```diff
@@ -555,9 +584,9 @@ def thread_run_job(
             exception = e
             wandb._sentry.exception(e)
         finally:
-            self.finish_thread_id(thread_id, exception)
+            await self.finish_thread_id(rqi_id, exception)
 
-    def _thread_run_job(
+    async def _thread_run_job(
```

Comment:
this feels like it should be renamed now?

Reply:
yes! will do